File: MiniDOMOffice/Office/RichiesteCERQ/MotiviRichieste/MotivoRichiestaCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;

namespace minidom
{
    public partial class Office
    {


        /// <summary>
        /// Repository di <see cref="MotivoRichiesta"/>
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class MotivoRichiestaCursor 
            : minidom.Databases.DBObjectCursor<MotivoRichiesta>
        {
            private DBCursorStringField m_Motivo = new DBCursorStringField("Motivo");
            private DBCursorField<MotivoRichiestaFlags> m_Flags = new DBCursorField<MotivoRichiestaFlags>("Flags");
            private DBCursorStringField m_HandlerName = new DBCursorStringField("NomeHandler");

            /// <summary>
            /// Costruttore
            /// </summary>
            public MotivoRichiestaCursor()
            { 
            }

            /// <summary>
            /// Motivo
            /// </summary>
            public DBCursorStringField Motivo
            {
                get
                {
                    return m_Motivo;
                }
            }

            /// <summary>
            /// Flags
            /// </summary>
            public DBCursorField<MotivoRichiestaFlags> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            /// <summary>
            /// HandlerName
            /// </summary>
            public DBCursorStringField HandlerName
            {
                get
                {
                    return m_HandlerName;
                }
            }

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Office.RichiesteCERQ.MotiviRichieste;
            }
             
        }
    }
}

File: MiniDOMOffice/Office/RichiesteCERQ/MotiviRichieste/MotivoRichiestaCursor.vb
Imports minidom.Databases
Imports minidom.Sistema

Partial Class Office


    ''' <summary>
    ''' Cursore sulla tabella dei motivi delle richieste
    ''' </summary>
    ''' <remarks></remarks>
    Public Class MotivoRichiestaCursor
        Inherits DBObjectCursor(Of MotivoRichiesta)

        Private m_Motivo As New CCursorFieldObj(Of String)("Motivo")
        Private m_Flags As New CCursorField(Of MotivoRichiestaFlags)("Flags")
        Private m_HandlerName As New CCursorFieldObj(Of String)("NomeHandler")

        Public ReadOnly Property Motivo As CCursorFieldObj(Of String)
            Get
                Return Me.m_Motivo
            End Get
        End Property

        Public ReadOnly Property Flags As CCursorField(Of MotivoRichiestaFlags)
            Get
                Return Me.m_Flags
            End Get
        End Property

        Public ReadOnly Property HandlerName As CCursorFieldObj(Of String)
            Get
                Return Me.m_HandlerName
            End Get
        End Property

        Protected Overrides Function GetModule() As CModule
            Return Office.RichiesteCERQ.MotiviRichieste.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_OfficeRichiesteM"
        End Function

        Protected Overrides Function GetConnection() As CDBConnection
            Return Office.Database
        End Function
    End Class



End Class